pkg/policy/api/fqdn.go

```diff
@@ -62,6 +62,9 @@ func (s *FQDNSelector) String() string {
 // when using MatchName the basic requirement is that is a valid regexp. We
 // test that it can compile here.
 func (s *FQDNSelector) sanitize() error {
+	if len(s.MatchName) > 0 && len(s.MatchPattern) > 0 {
+		return fmt.Errorf("only one of MatchName or MatchPattern is allowed in an FQDNSelector")
```

Comment:
This is surprising to me.. Is there a reason why we can't have both? cc @raybejjani

Reply:
The rationale behind this is that with the upcoming usage of `SelectorCache`, we will have a mapping of a selector --> identities which match that selector. In the case of FQDN, the fact that the same selector can have two disjoint fields which may match completely separate things does not make sense conceptually - say I have `MatchName: google.com` and then `MatchPattern: *.cilium.io` - both are completely disjoint. I want to be able to map a given FQDNSelector to the identities which match that selector - if we have two fields which are disjoint in the same selector, we lose a lot of granularity. Also, I will be using the `FQDNSelector` as a key in a map and mapping it to a regular expression representing the selector, so having both fields populated would make such behavior impossible.